### PR TITLE
Add better support for AppOptions with multiple types (PR 18465 follow-up)

### PR DESCRIPTION
**Pull request by Snuffleupagus**
For options with varying types, see `useSystemFonts`, we're not sufficiently validating the type when setting a new value. This means that for an option that uses `OptionKind.UNDEF_ALLOWED` we'd allow *any* value, which is obviously not the intention.

Hence we instead introduce a new and *optional* `type`-field that allows specifying exactly which types are valid when multiple ones are supported.

*Note:* This obviously didn't occur to me until after PR #18465 landed, sorry about that!

### CODE REVIEW

File: web/app_options.js [hunk 1]
```diff
@@ -55,10 +55,19 @@ const OptionKind = {
   API: 0x04,
   WORKER: 0x08,
   EVENT_DISPATCH: 0x10,
-  UNDEF_ALLOWED: 0x20,
   PREFERENCE: 0x80,
 };
 
+// These fields *must* be lowercase to match the `typeof` operator,
```
Comment (timvandermeij) [inline web/app_options.js:61]: Alternatively we could also use `(typeof value).toUpperCase()` below instead. This allows us to remove the comment and use uppercased enum keys as usual.

**Review by timvandermeij — APPROVED** [review]
r=me, with one optional comment. Thank you for improving this!

File: web/app_options.js (hunk 1 above)
Comment (Snuffleupagus) [reply web/app_options.js]: That's a really good suggestion, since the inconsistency probably isn't great; thank you!

**Pull request merged**